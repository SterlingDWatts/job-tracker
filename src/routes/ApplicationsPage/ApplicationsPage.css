.ApplicationsPage .columns {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  padding: 16px;
  overflow-x: scroll;
  -ms-overflow-style: none;  /* IE and Edge */
  scrollbar-width: none;  /* Firefox */
}

.ApplicationsPage .columns::-webkit-scrollbar {
  display: none;
}