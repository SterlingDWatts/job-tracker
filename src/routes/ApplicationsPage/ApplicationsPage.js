import React, { useState } from "react";
import { Route } from "react-router-dom";
import Column from "../../components/Column/Column";
import LargeCard from "../../components/LargeCard/LargeCard";
import "./ApplicationsPage.css";

const jobApps = [
  {
    id: 0,
    column: [{ id: 0, date: new Date(2020, 4, 13) }],
    company: "Ultimate Software",
    job: "Sotware Engineer 2",
    site: "https://www.ultimatesoftware.com/",
  },
  {
    id: 1,
    column: [
      { id: 0, date: new Date(2020, 4, 14) },
      { id: 3, date: new Date(2020, 4, 14) },
    ],
    company: "Art & Logic",
    job: "Full Stack Web Developer",
    site: "https://artandlogic.com/",
  },
  {
    id: 2,
    column: [{ id: 0, date: new Date(2020, 4, 14) }],
    company: "GitStart",
    job: "Full Stack JavaScript Developer - Junior",
    site: "https://www.gitstart.com/",
  },
  {
    id: 3,
    column: [{ id: 0, date: new Date(2020, 4, 14) }],
    company: "CyberCoders",
    job: "Web Developer",
    site: "https://www.cybercoders.com/",
  },
  {
    id: 4,
    column: [{ id: 0, date: new Date(2020, 4, 14) }],
    company: "Digital Media Solutions",
    job: "Frontend Developer",
    site: "https://digitalmediasolutions.com/",
  },
  {
    id: 5,
    column: [{ id: 0, date: new Date(2020, 11, 1) }],
    company: "Fake",
    job: "Fake",
    site: "https://www.google.com",
  },
];

export default function ApplicationsPage(props) {
  const [applications, setApplications] = useState(jobApps);
  const [columns, setColumns] = useState([
    { id: 0, name: "Applied" },
    { id: 1, name: "Intro Interview" },
    { id: 2, name: "Recruiter Call" },
    { id: 3, name: "Sent Follow-up" },
    { id: 4, name: "Coding Challenge" },
    { id: 5, name: "Sent Follow-up" },
    { id: 6, name: "Technical Interview" },
    { id: 7, name: "Sent Follow-up" },
    { id: 8, name: "Onsite Interview" },
    { id: 9, name: "Offer Letter Date" },
  ]);
  const columnComponents = columns.map((col) => (
    <Column
      name={col.name}
      key={col.id}
      id={col.id}
      applications={applications}
      setApplications={setApplications}
    />
  ));
  return (
    <div className="ApplicationsPage">
      <div className="columns">{columnComponents}</div>
      <Route path="/application/:appId">
        <LargeCard
          applications={applications}
          setApplications={setApplications}
          columns={columns}
        />
      </Route>
    </div>
  );
}
