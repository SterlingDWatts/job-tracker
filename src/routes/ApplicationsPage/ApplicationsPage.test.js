import React from "react";
import ReactDOM from "react-dom";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import ApplicationsPage from "./ApplicationsPage";

describe("ApplicationsPage Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <ApplicationsPage />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = render(
      <MemoryRouter initialEntries={["/?app=1"]}>
        <ApplicationsPage />
      </MemoryRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
