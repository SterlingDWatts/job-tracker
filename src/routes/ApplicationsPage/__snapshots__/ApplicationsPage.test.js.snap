// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationsPage Component renders the UI as expected 1`] = `
<div
  className="ApplicationsPage"
>
  <div
    className="top"
  >
    <Link
      className="add-app"
      to="/add-application"
    >
      Add
    </Link>
  </div>
  <div
    className="column-headers"
  >
    <div
      key="0"
    >
      Applied
    </div>
    <div
      key="1"
    >
      Intro Interview
    </div>
    <div
      key="2"
    >
      Recruiter Call
    </div>
    <div
      key="3"
    >
      Sent Follow-up
    </div>
    <div
      key="4"
    >
      Coding Challenge
    </div>
    <div
      key="5"
    >
      Sent Follow-up
    </div>
    <div
      key="6"
    >
      Technical Interview
    </div>
    <div
      key="7"
    >
      Sent Follow-up
    </div>
    <div
      key="8"
    >
      Onsite Interview
    </div>
    <div
      key="9"
    >
      Offer Letter Date
    </div>
  </div>
  <div
    className="columns"
  >
    <Column
      id={0}
      key="0"
    />
    <Column
      id={1}
      key="1"
    />
    <Column
      id={2}
      key="2"
    />
    <Column
      id={3}
      key="3"
    />
    <Column
      id={4}
      key="4"
    />
    <Column
      id={5}
      key="5"
    />
    <Column
      id={6}
      key="6"
    />
    <Column
      id={7}
      key="7"
    />
    <Column
      id={8}
      key="8"
    />
    <Column
      id={9}
      key="9"
    />
  </div>
  <Route
    path="/application/:appId"
  >
    <LargeCard
      columns={
        Array [
          Object {
            "id": 0,
            "name": "Applied",
          },
          Object {
            "id": 1,
            "name": "Intro Interview",
          },
          Object {
            "id": 2,
            "name": "Recruiter Call",
          },
          Object {
            "id": 3,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 4,
            "name": "Coding Challenge",
          },
          Object {
            "id": 5,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 6,
            "name": "Technical Interview",
          },
          Object {
            "id": 7,
            "name": "Sent Follow-up",
          },
          Object {
            "id": 8,
            "name": "Onsite Interview",
          },
          Object {
            "id": 9,
            "name": "Offer Letter Date",
          },
        ]
      }
    />
  </Route>
  <Route
    path="/add-application"
  >
    <AddApplication />
  </Route>
</div>
`;
